Mathlib/Analysis/SpecialFunctions/Trigonometric/Bounds.lean

```diff
@@ -53,6 +56,64 @@ theorem sin_lt {x : ℝ} (h : 0 < x) : sin x < x := by
   norm_num
 #align real.sin_lt Real.sin_lt
 
+lemma cos_quadratic_lower_bound : 1 - x ^ 2 / 2 ≤ cos x := by
+  wlog hx₀ : 0 ≤ x
+  · simpa using this $ neg_nonneg.2 $ le_of_not_le hx₀
+  suffices : MonotoneOn (fun x ↦ cos x + x ^ 2 / 2) (Ici 0)
+  · simpa using this left_mem_Ici hx₀ hx₀
+  refine monotoneOn_of_hasDerivWithinAt_nonneg (convex_Ici _) (Continuous.continuousOn $ by
+    continuity) (fun x _ ↦
+    ((hasDerivAt_cos ..).add $ (hasDerivAt_pow ..).div_const _).hasDerivWithinAt) fun x hx ↦ ?_
+  simp [mul_div_cancel_left]
+  exact sin_le $ interior_subset hx
+
+/-- **Jordan's inequality**. -/
+lemma two_div_pi_mul_le (hx₀ : 0 ≤ x) (hx : x ≤ π / 2) : 2 / π * x ≤ sin x := by
+  rw [← sub_nonneg]
+  suffices : ConcaveOn ℝ (Icc 0 (π / 2)) (fun x ↦ sin x - 2 / π * x)
+  · refine (le_min ?_ ?_).trans $ this.min_le_of_mem_Icc ⟨hx₀, hx⟩ <;> field_simp
+  refine concaveOn_of_hasDerivWithinAt2_nonpos (convex_Icc ..)
+    (Continuous.continuousOn $ by continuity)
+    (fun x _ ↦ ((hasDerivAt_sin ..).sub $ (hasDerivAt_id ..).const_mul (2 / π)).hasDerivWithinAt)
+    (fun x _ ↦ (hasDerivAt_cos ..).hasDerivWithinAt.sub_const _)
+    fun x hx ↦ neg_nonpos.2 $ sin_nonneg_of_mem_Icc $ Icc_subset_Icc_right (by linarith) $
+    interior_subset hx
+
+lemma cos_quadratic_upper_bound (hx : |x| ≤ π) : cos x ≤ 1 - 2 / π ^ 2 * x ^ 2 := by
+  wlog hx₀ : 0 ≤ x
+  · simpa using this (by rwa [abs_neg]) $ neg_nonneg.2 $ le_of_not_le hx₀
+  rw [abs_of_nonneg hx₀] at hx
+  -- TODO: `compute_deriv` tactic?
+  have hderiv (x) : HasDerivAt (fun x ↦ 1 - 2 / π ^ 2 * x ^ 2 - cos x) _ x :=
+    (((hasDerivAt_pow ..).const_mul _).const_sub _).sub $ hasDerivAt_cos _
+  simp only [Nat.cast_ofNat, Nat.succ_sub_succ_eq_sub, tsub_zero, pow_one, ← neg_sub', neg_sub,
+    ← mul_assoc] at hderiv
+  have hmono : MonotoneOn (fun x ↦ 1 - 2 / π ^ 2 * x ^ 2 - cos x) (Icc 0 (π / 2))
+  · refine monotoneOn_of_hasDerivWithinAt_nonneg (convex_Icc ..) (Continuous.continuousOn $
+      by continuity) (fun x _ ↦ (hderiv _).hasDerivWithinAt) fun x hx ↦ sub_nonneg.2 ?_
+    have ⟨hx₀, hx⟩ := interior_subset hx
+    calc
+      _ = 2 / π * (2 / π * x) := by ring
```

Comment:
When you start a calc block, can you write down explicitly the starting term instead of just `_`, to help the reader?

Reply:
It makes the indentation all wrong, but sure